File: py_test/second/cov_plot_calcRoee_test.py
import numpy as np
import csv
from cov_plot_calcRoee import cov_plot_calcRoee  # Assuming you've saved your function in this file
import math


def load_csv_to_array(file_path):
    # Load the CSV file into a NumPy array
    data = np.genfromtxt(file_path, delimiter=',')
    
    # Get the size (number of elements) in the array
    array_size = data.size
    
    return data, array_size


def load_csv_to_2d_array(file_path):
    # Load the CSV file into a NumPy array
    data = np.genfromtxt(file_path, delimiter=',')
    
    # Get the shape of the array to determine if it's 1D or 2D
    if data.ndim == 1:
        # If it's 1D, reshape to 2D with one row
        data = data.reshape(1, -1)
    
    # Get the size (number of elements) in the array
    array_size = data.size
    
    return data, array_size



def test_cov_plot_calcRoee():
    # Test parameters
    # Load inputs from CSV files (assuming r, teta, and sigmaTeta are stored in CSV files)
    MergedRng, _ = load_csv_to_array('C:/Users/ipewz/Desktop/roee_proj/forTest/cov_plot_calcRoee/MergedRng.csv')
    MergedTeta, _ = load_csv_to_array('C:/Users/ipewz/Desktop/roee_proj/forTest/cov_plot_calcRoee/MergedTeta.csv')
    #rc, _ = load_csv_to_1d_array('C:\Users\ipewz\Desktop\roee_proj\forTest\cov_plot_calcRoee\Rc.csv')
    sigmaTeta=3
    k=0
    # For simplicity, assuming r, teta, sigmaTeta are scalars

    # Call the cov_plot_calcRoee function
    calculated_rc = cov_plot_calcRoee(MergedRng[k], MergedTeta[k]*math.pi/180, sigmaTeta)

    # Define expected result (2x2 matrix)
    #expected_rc = np.array([[223.9,-3.9384], [-3.9384,223.9]])  # Replace with actual expected values
    expected_rc = np.array([[223.8971507800168865,-3.9383795899505540], [
-3.9383795899505540,223.8971507792297189]])  # Replace with actual expected values

    # Use np.allclose to compare matrices
    assert np.allclose(calculated_rc, expected_rc, rtol=1e-9), \
        f"calculated_rc {calculated_rc} does not match expected {expected_rc}"

    print("calculated_rc:", calculated_rc)
    print("expected_rc:", expected_rc)

if __name__ == "__main__":
    test_cov_plot_calcRoee()
    print("Test passed")

File: c/secund/cov_plot_calcRoee_test.c

#include "cov_plot_calcRoee.h"

#include <assert.h>
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#define PI 3.14159

void test_cov_plot_calcRoee() {
    // Load inputs from CSV files
    double *MergedRng, *MergedTeta;
    size_t sizeRng, sizeTeta;


    double sigmaTeta = 3.0;
    size_t k = 0;  // Assuming we are testing the first element

    double rc[4];
    cov_plot_calcRoee(21.073242187500000, 1.125457163528671, 3, rc);
    double expected_rc[4] = { 2.221033605856399e+02,
                              -8.595739415541884,
                              -8.595739415541884,
                              2.221033605851150e+02 };
   
    // Compare matrices
    for (int i = 0; i < 4; i++) {
        //printf("rc[%d] = %f, expected_rc[%d] = %f\n", i, rc[i], i, expected_rc[i]);
        if (fabs(rc[i] - expected_rc[i]) < 1e-3||fabs(rc[i] - expected_rc[i]) > 1e-3) {
        printf("\nfabs(rc[i] - expected_rc[i]%f",(fabs(rc[i] - expected_rc[i]))); 
        }
    }
    printf("\nTest passed\n");

}

int main() {
    test_cov_plot_calcRoee();
    return 0;
}

/*

 gcc cov_plot_calcRoee_test.c -o cov_plot_calcRoee_test.exe -lm
./cov_plot_calcRoee_test.exe

*/